code/modules/mob/death.dm

```diff
@@ -76,6 +76,12 @@
 	for(var/obj/item/implant/carrion_spider/control/C in src)
 		C.return_mind()
 
+	// Notifies borer of host death.
+	if(has_brain_worms())
+		var/mob/living/simple_animal/borer/B = has_brain_worms()
+		if(B?.controlling)
+			B.host_death()
```

Comment:
```suggestion
	var/mob/living/simple_animal/borer/B = has_brain_worms()
	if(B?.controlling)
		B.host_death()
```
Sorry I should have been more clear, it's cleaner by having has_brain_worms() return `null` instead of a weird function that returns either an object or FALSE. We want either `borer/null`  or `TRUE/FALSE`. I made a PR for that #7967 and will merge yours after.